### ISSUE TITLE
Failed create pod sandbox: rpc error: code = Unknown desc

### ISSUE THREAD

**Issue by itsecforu** [coredns/deployment#198]
coredns pod doenst start

` kubectl get pods --all-namespaces
NAMESPACE     NAME                                       READY   STATUS              RESTARTS   AGE
kube-system   calico-kube-controllers-6d57b44787-xlj89   1/1     Running             15         10d
kube-system   calico-node-dwm47                          1/1     Running             310        5d3h
kube-system   calico-node-hhgzk                          1/1     Running             13         10d
kube-system   calico-node-tk4mp                          1/1     Running             309        5d3h
kube-system   calico-node-w7zvs                          1/1     Running             311        5d3h
kube-system   coredns-74c9d4d795-psf2v                   1/1     Running             0          4d4h
**kube-system   coredns-74c9d4d795-xpbsd                   0/1     ContainerCreating**   0          5d3h
kube-system   dns-autoscaler-7d95989447-7kqsn            1/1     Running             8          10d
kube-system   kube-apiserver-master                      1/1     Running             1          10d
kube-system   kube-controller-manager-master             1/1     Running             1          10d
kube-system   kube-proxy-9bt8m                           1/1     Running             1          5d3h
kube-system   kube-proxy-cbrcl                           1/1     Running             2          5d3h
kube-system   kube-proxy-stj5g                           1/1     Running             0          5d3h
kube-system   kube-proxy-zql86                           1/1     Running             0          5d3h
kube-system   kube-scheduler-master                      1/1     Running             1          10d
kube-system   kubernetes-dashboard-7c547b4c64-6skc7      1/1     Running             589        10d
kube-system   nginx-proxy-worker1                        1/1     Running             1          5d3h
kube-system   nginx-proxy-worker2                        1/1     Running             0          5d3h
kube-system   nginx-proxy-worker3                        1/1     Running             0          5d3h
kube-system   nodelocaldns-6t92x                         1/1     Running             1          5d3h
kube-system   nodelocaldns-kgm4t                         1/1     Running             0          5d3h
kube-system   nodelocaldns-xl8zg                         1/1     Running             0          5d3h
kube-system   nodelocaldns-xwlwk                         1/1     Running             10         10d
`
OS:

```
cat /etc/os-release
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7
```

my inventory:

```
master ansible_host=10.2.62.251 ansible_user=root
worker1 ansible_host=10.2.62.253 ansible_user=root
worker2 ansible_host=10.2.62.255 ansible_user=root
worker3 ansible_host=10.2.62.256 ansible_user=root
#[all:vars]
#ansible_python_interpreter=/usr/bin/python3

[kube-master]
master

[kube-node]
worker1
worker2
worker3

[etcd]
master

[calico-rr]

[k8s-cluster:children]
kube-master
kube-node
```

Log :

`Failed create pod sandbox: rpc error: code = Unknown desc = failed to start sandbox container for pod "coredns-74c9d4d795-xpbsd": Error response from daemon: OCI runtime create failed: container_linux.go:345: starting container process caused "process_linux.go:303: getting the final child's pid from pipe caused \"read init-p: connection reset by peer\"": unknown
`
![1](https://user-images.githubusercontent.com/38696837/65428431-09b84580-de1d-11e9-977b-b7487b9c2aa9.jpg)

**Comment by chrisohaver** [comment]
looks like a docker problem

**Comment by itsecforu** [comment]
``` 
docker version
Client: Docker Engine - Community
 Version:           19.03.2
 API version:       1.39 (downgraded from 1.40)
 Go version:        go1.12.8
 Git commit:        6a30dfc
 Built:             Thu Aug 29 05:28:55 2019
 OS/Arch:           linux/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          18.09.7
  API version:      1.39 (minimum version 1.12)
  Go version:       go1.10.8
  Git commit:       2d0083d
  Built:            Thu Jun 27 17:26:28 2019
  OS/Arch:          linux/amd64
  Experimental:     false
```

**Comment by chrisohaver** [comment]
can you `sudo docker run --rm  -it hello-world` on the same node that the coredns pod cannot start?

**Comment by itsecforu** [comment]
```
sudo docker run --rm -it hello-world
Unable to find image 'hello-world:latest' locally
docker: Error response from daemon: Get https://registry-1.docker.io/v2/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers).
See 'docker run --help'.
```

**Comment by chrisohaver** [comment]
So that node cannot reach the outside world?

**Comment by itsecforu** [comment]
Hmm , I can ping google:
```
 ping 8.8.8.8
PING 8.8.8.8 (8.8.8.8) 56(84) bytes of data.
64 bytes from 8.8.8.8: icmp_seq=1 ttl=47 time=17.5 ms
64 bytes from 8.8.8.8: icmp_seq=2 ttl=47 time=17.2 ms
64 bytes from 8.8.8.8: icmp_seq=3 ttl=47 time=19.4 ms
64 bytes from 8.8.8.8: icmp_seq=4 ttl=47 time=18.7 ms
```

**Comment by chrisohaver** [comment]
It seems that at least docker cannot reach the docker registry.  Maybe thats blocked on your network.  Does ... `docker run --rm -it k8s.gcr.io/coredns:1.6.2` work?

**Comment by itsecforu** [comment]
seems as nope :-(

```
docker run --rm -it k8s.gcr.io/coredns:1.6.2
Unable to find image 'k8s.gcr.io/coredns:1.6.2' locally
docker: Error response from daemon: Get https://k8s.gcr.io/v2/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers).
See 'docker run --help'.
```

**Comment by itsecforu** [comment]
tryed to make:

make a file:
```
vi /etc/sysconfig/docker
```
with:
```
HTTP_PROXY="http://mylocal-squid:3128"
HTTPS_PROXY="http://mylocal-squid:3128"
http_proxy="${HTTP_PROXY}"
https_proxy="${HTTPS_PROXY}"
```
Then restart Docker:

```
service docker restart
```
Doesnt work(

**Comment by itsecforu** [comment]
Is it normal that the coredns works on the master node and on worker1, although there are three workers, should it not only be on the master as a service service?
![1](https://user-images.githubusercontent.com/38696837/65492293-57828b80-deb9-11e9-9414-d0270950491c.jpg)

**Comment by chrisohaver** [comment]
> Is it normal that the coredns works on the master node and on worker1, although there are three workers

Yes. This can happen if your master doesn't have the [no-schedule taint](https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/create-cluster-kubeadm/#control-plane-node-isolation) so it accepts scheduled pods.  It also will happen if your coredns Deployment is configured to tolerate (ignore) the no-schedule taint.

> docker: Error response from daemon

Looks like docker is broken on worker1.  
Try testing docker connectivity from other nodes to see if they suffer the same issue.  You may want to make sure you dont have a firewall somewhere blocking https.

I'm not adept at diagnosing and resolving docker issues. But I would try rebooting the affected node to see if that helps. If that doesn't help maybe try uninstalling/re-installing docker server and client, or rebuilding the node.

**Comment by itsecforu** [comment]
thx for feedback.

the same erorror on others nodes.

And I found this solution https://medium.com/@dirk.avery/docker-error-response-from-daemon-1d46235ff61d 

But I do not want to risk and update the kernel

but it does not work's , if even I stop firewalld.

**Comment by harshakondreddy** [comment]
**Getting below issue**
`Failed to create pod sandbox: rpc error: code = Unknown desc = failed to inspect sandbox image "k8s.gcr.io/pause:3.2": Error response from daemon: layer does not exist`

evaluated by 
```
sudo docker run --rm -it hello-world
# got this error#
Status: Downloaded newer image for hello-world:latest
docker: Error response from daemon: failed to update store for object type *libnetwork.endpointCnt: Key not found in store.
```
 Restarting docker worked for me...
`Systemctl restart docker`
